### ISSUE TITLE
Activity feed shows new users join who are not part of the current user's group

### ISSUE THREAD

**Issue by hagen** [metabase/metabase#6184]
- Your browser and the version: Chrome Version 61.0.3163.100
- Your operating system: Win 10
- Your databases: AWS Aurora (MySQL)
- Metabase version: v0.25.2
- Metabase hosting environment: Elastic Beanstalk
- Metabase internal database: Postgres
- Repeatable steps to reproduce the issue

1. Admin creates a new group, Group A
2. Admin creates a new group, Group B
3. Admin creates new user, User A, adds to Group A and All Users (default).
4. Admin creates new user, User B, adds to Group B and All Users (default).
5. Log out, refresh browser, log in as User A.
6. Log out, refresh browser, log in as User B. User B will see in their activity feed that User A joined.

I would have expected not to see User A joining in User B's activity feed, as User A is in a different Group. Groups are used to segregate access to Metabase resources - Databases/Schemas/Tables. Seems that different groups should know nothing about the users in other groups?

**Comment by doanvelagui** [comment]
Same here, I would like to have group activity feeds isolated (regarding _user-join_ events). I guess this _issue_ is "produced" since all users belongs to _All Users_ group. If you have permissions to a _metabase resource_ for _All Users_ group, everybody [in this group] can see in their feed actions related to this _metabase resource_.

As a workaround (I have my metabase database in postgres) I deleted this _user-joined_ entry from table _activity_ manually (now no one can see it). I do not want to remove memberships or _All Users_ group because don't know which effect can have, but I'm sure there are better/more-elegant solutions, just be creative.


```
-- just to get the ID and ensure I am deleting the correct one
select * from activity order by id desc limit 1;
 id  |    topic    |         timestamp          | user_id | model | model_id | database_id | table_id | custom_id | details
-----+-------------+----------------------------+---------+-------+----------+-------------+----------+-----------+---------
 644 | user-joined | 2017-10-17 16:58:16.512+00 |      43 | user  |       43 |             |          |           | {}
(1 row)

-- in my case, the ID is 644
delete from activity where id = 644;
DELETE 1
```

**Comment by hagen** [comment]
@doanvelagui thanks, I’ve done that too. this work around has to be done each time a new user joins. And the activity event is only recorded when the user logs in for the first time, which I don’t necessarily know, so the record hangs around for some time until I delete it.

The crux is - as you say - the _resource_ user-joined is not part of the permissions graph/matrix and therefore has no special permissions relating to it. So every user will see it in their activity feed.

**Comment by doanvelagui** [comment]
Yes, it is true that you don't know when a new user will join. Probably, something like this will be more helpful:

```
create or replace function activity_insert() returns trigger as $$
    begin
        if new.topic = 'user-joined' then
            return null;
        else
            return new;
        end if;
    end
$$ language plpgsql;

create trigger user_joined before insert
    on activity
    for each row execute procedure activity_insert();
```

**Comment by hagen** [comment]
Yeh! I was just thinking a stored proc/trigger is the way to go.

**Comment by hagen** [comment]
Yeh! I was just thinking a stored proc/trigger is the way to go.

**Comment by hagen** [comment]
Left the issue open as I think 'user-joins' activities need to be restricted by group - if the new user is not in your Group, you shouldn't see them joining in your activity feed.

**Comment by flamber** [comment]
I cannot reproduce on 0.35.3 - think this was fixed in 0.30, with the big Collection change.
